packages/strapi-design-system/src/Stack/Stack.tsx

```diff
@@ -1,42 +1,60 @@
-import React, { forwardRef } from 'react';
-import PropTypes from 'prop-types';
-import styled from 'styled-components';
-import { Flex } from '../Flex';
+import { forwardRef } from 'react';
+import styled, { DefaultTheme } from 'styled-components';
+
+import { Flex, FlexProps } from '../Flex';
+import { extractStyleFromTheme } from '../helpers/theme';
 
 /**
  * Prevents these attributes from being spread on the DOM node
  */
-const transientProps = {
+const transientProps: Partial<Record<keyof StackProps, boolean>> = {
   size: true,
 };
 
-const StackV = styled(Flex).withConfig({
-  shouldForwardProp: (prop, defPropValFN) => !transientProps[prop] && defPropValFN(prop),
+const StackV = styled(Flex).withConfig<Omit<StackProps, 'size' | 'horizontal'>>({
+  shouldForwardProp: (prop, defPropValFN) => !transientProps[prop as keyof StackProps] && defPropValFN(prop),
 })`
   & > * {
     margin-top: 0;
     margin-bottom: 0;
   }
 
   & > * + * {
-    margin-top: ${({ theme, spacing }) => theme.spaces[spacing]};
+    margin-top: ${({ theme, spacing }) => extractStyleFromTheme(theme.spaces, spacing, undefined)};
```

Comment:
I guess calling `extractStyleFromTheme()` is necessary to TS to infer the proper type, right? I'm wondering if it would be possible to implement this with getter methods or converting `theme.*` to proxy objects? This way we could keep the boilerplate out of styles and centralize the type inference?

Reply:
Did you have an API in mind? I'm not super familiar with Proxy objects.

Reply:
Something like:

```js
const spaces = [1,2,3];
const spacesProxy = {
  get(target, index) /* infer type */ {
    target[index];
  }
}

theme.spaces = new Proxy(spaces, spacesProxy);
```

This is just an idea: I have no idea if this could work ...

Reply:
I'm gonna open this a separate issue to not block the progression, I think it could be a neat idea and cleaner – you're totally right.